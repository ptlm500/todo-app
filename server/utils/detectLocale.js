const acceptLanguage = require('accept-language');
const config = require('../config.json');

function detectLocale(req) {
  acceptLanguage.languages(config.ACCEPTED_LANGUAGES);

  const cookieLocale = req.cookies.locale;

  return acceptLanguage.get(cookieLocale || req.headers['accept-language']) ||
    config.DEFAULT_LOCALE;
}

module.exports = detectLocale;
