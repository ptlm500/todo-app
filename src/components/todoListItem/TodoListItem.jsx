import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { removeTodo } from '../../state/actions/actions';

export function TodoListItem(props) {
  const {
    id,
    name,
    description,
    creationDate,
    onRemove
  } = props;

  function remove() {
    onRemove(id);
  }

  return (
    <li className='todo-list__item'>
      <div>
        <h3>{name}</h3>
        <button onClick={remove}>
          Remove
        </button>
      </div>
      <a>{description}</a>
      <a>{creationDate}</a>
    </li>
  );
}

TodoListItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  creationDate: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => {
  return {
    onRemove: id => dispatch(removeTodo(id))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(TodoListItem);
