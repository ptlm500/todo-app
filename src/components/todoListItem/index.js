export { default } from './TodoListItem.jsx';
