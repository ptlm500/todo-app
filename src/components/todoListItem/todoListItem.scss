@import '../../vars';

.todo-list__item {
  display: flex;
  flex-direction: column;
  background-color: $element-01;
  margin: 1rem;
  padding: 0.5rem;
  border-radius: 0.5rem;
}
