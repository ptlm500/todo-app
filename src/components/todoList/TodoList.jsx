import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import { connect } from 'react-redux';
import TodoListItem from '../todoListItem';

function renderTodos(todos) {
  return Object.values(todos)
    .map((todo) => <TodoListItem key={todo.id} {...todo} />);
}


export function TodoList(props) {
  const { todos } = props;

  return (
    <ul>
      {renderTodos(todos)}
    </ul>
  );
}

TodoList.propTypes = {
  todos: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    creationDate: PropTypes.string.isRequired
  }))
};

export function Container(props) {
  const { todos } = props;

  return (
    <TodoList todos={todos.toJS && todos.toJS()} />
  );
}

Container.propTypes = {
  todos: PropTypes.instanceOf(Immutable.Map)
};

const mapStateToProps = (state) => ({
  todos: state.todos
});


export default connect(
  mapStateToProps
)(Container);
