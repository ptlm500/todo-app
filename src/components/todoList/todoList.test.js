import React from 'react';
import { fromJS } from 'immutable';
import { Provider } from 'react-redux';
import store from '../../state/store';
import { renderWithIntl } from '../../../test/utils';
import { StyledTodoList, Container } from './TodoList';
import { TodoListItem } from '../todoListItem/TodoListItem';

// Mocking transition group https://github.com/reactjs/react-transition-group/issues/208
jest.mock('react-transition-group', () => {
  return {
    // eslint-disable-next-line react/prop-types
    CSSTransition: function CSSTransition({children}) {
      return (
        <div>{children}</div>
      );
    }
  };
});

const testTodos = {
  'test-id': {
    id: 'test-id',
    name: 'Test Todo',
    description: 'Pass all tests!',
    creationDate: '1581549548000'
  },
  'test-id-2': {
    id: 'test-id-2',
    name: 'Test Todo 2',
    description: 'Especially this test!',
    creationDate: '1581549548000'
  }
};

describe('TodoList', () => {
  it('renders the correct number of todos, matching the snapshot', () => {
    const tree = renderWithIntl(
      <Provider store={store}>
        <StyledTodoList todos={testTodos} />
      </Provider>
    );

    const todoItems = tree.root.findAllByType(TodoListItem);

    const testTodoValues = Object.values(testTodos);
    expect(todoItems.length).toBe(testTodoValues.length);

    todoItems.forEach((item, idx) => {
      expect(item.props.id).toBe(testTodoValues[idx].id);
      expect(item.props.name).toBe(testTodoValues[idx].name);
      expect(item.props.description).toBe(testTodoValues[idx].description);
      expect(item.props.creationDate).toBe(testTodoValues[idx].creationDate);
    });

    expect(tree).toMatchSnapshot();
  });
});

describe('container', () => {
  it('converts immutable todos into a js object', () => {
    const tree = renderWithIntl(
      <Provider store={store}>
        <Container todos={fromJS(testTodos)} />
      </Provider>
    );

    const styledTodoList = tree.root.findByType(StyledTodoList);
    expect(styledTodoList.props.todos).toEqual(testTodos);
  });
});
