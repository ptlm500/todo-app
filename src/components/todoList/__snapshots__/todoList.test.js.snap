// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders the correct number of todos, matching the snapshot 1`] = `
.c7 {
  font-size: 2.5vw;
  color: #757575;
}

.c4 {
  white-space: pre-wrap;
  word-break: break-word;
}

.c6 {
  font-size: 2.5vw;
  margin-bottom: 1rem;
  min-width: 100px;
}

.c5 {
  background: none;
  border: none;
  width: 1rem;
  height: 1rem;
  padding: 0;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c5 > svg {
  width: 100%;
  height: 100%;
}

.c5:hover {
  -webkit-transform: scale(1.2) rotate(1turn);
  -ms-transform: scale(1.2) rotate(1turn);
  transform: scale(1.2) rotate(1turn);
}

.c3 {
  font-size: 5vw;
  font-weight: 600;
  min-width: 100px;
}

.c1 {
  background: #C5CAE9;
  border-radius: 0.5rem;
  padding: 1rem;
  border: 1px solid #3F51B5;
  min-width: 450px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
}

.c1.card-transition-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c1.card-transition-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c1.card-transition-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c1.card-transition-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c1.card-transition-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c1.card-transition-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c1.card-transition-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c1.card-transition-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1rem;
}

.c0 {
  border-left: 1px solid #BDBDBD;
  padding-left: 0.5rem;
}

.c0 > * {
  margin: 1rem 0;
}

@media (max-width:480px) {
  .c7 {
    font-size: 12px;
  }
}

@media (min-width:640px) {
  .c7 {
    font-size: 16px;
  }
}

@media (max-width:480px) {
  .c6 {
    font-size: 12px;
  }
}

@media (min-width:640px) {
  .c6 {
    font-size: 16px;
  }
}

@media (max-width:240px) {
  .c3 {
    font-size: 12px;
  }
}

@media (min-width:640px) {
  .c3 {
    font-size: 32px;
  }
}

@media (max-width:599px) {
  .c1 {
    min-width: 300px;
    max-width: 90%;
  }
}

<div
  className="c0"
>
  <div>
    <div
      className="sc-AxjAm sc-AxirZ c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3 inline-editable"
        >
          <div
            className="inline-editable__edit-trigger"
            onClick={[Function]}
          >
            <span
              className="c4"
            >
              Test Todo
            </span>
          </div>
        </div>
        <button
          className="c5"
          onClick={[Function]}
          size="small"
        >
          <svg
            height="18px"
            version="1.1"
            viewBox="0 0 14 18"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Delete
            </title>
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <g
                transform="translate(-207.000000, -245.000000)"
              >
                <g
                  transform="translate(100.000000, 100.000000)"
                >
                  <g
                    id="Outlined-/-Action-/-delete"
                    transform="translate(102.000000, 142.000000)"
                  >
                    <g>
                      <polygon
                        id="Path"
                        points="0 0 24 0 24 24 0 24"
                      />
                      <path
                        d="M16,9 L16,19 L8,19 L8,9 L16,9 Z M14.5,3 L9.5,3 L8.5,4 L5,4 L5,6 L19,6 L19,4 L15.5,4 L14.5,3 Z M18,7 L6,7 L6,19 C6,20.1 6.9,21 8,21 L16,21 C17.1,21 18,20.1 18,19 L18,7 Z"
                        fill="currentColor"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </button>
      </div>
      <div
        className="c6 inline-editable"
      >
        <div
          className="inline-editable__edit-trigger"
          onClick={[Function]}
        >
          <span
            className="c4"
          >
            Pass all tests!
          </span>
        </div>
      </div>
      <a
        className="c7"
      >
        February 12, 2020, 11:19 PM
      </a>
    </div>
  </div>
  <div>
    <div
      className="sc-AxjAm sc-AxirZ c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3 inline-editable"
        >
          <div
            className="inline-editable__edit-trigger"
            onClick={[Function]}
          >
            <span
              className="c4"
            >
              Test Todo 2
            </span>
          </div>
        </div>
        <button
          className="c5"
          onClick={[Function]}
          size="small"
        >
          <svg
            height="18px"
            version="1.1"
            viewBox="0 0 14 18"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Delete
            </title>
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <g
                transform="translate(-207.000000, -245.000000)"
              >
                <g
                  transform="translate(100.000000, 100.000000)"
                >
                  <g
                    id="Outlined-/-Action-/-delete"
                    transform="translate(102.000000, 142.000000)"
                  >
                    <g>
                      <polygon
                        id="Path"
                        points="0 0 24 0 24 24 0 24"
                      />
                      <path
                        d="M16,9 L16,19 L8,19 L8,9 L16,9 Z M14.5,3 L9.5,3 L8.5,4 L5,4 L5,6 L19,6 L19,4 L15.5,4 L14.5,3 Z M18,7 L6,7 L6,19 C6,20.1 6.9,21 8,21 L16,21 C17.1,21 18,20.1 18,19 L18,7 Z"
                        fill="currentColor"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
        </button>
      </div>
      <div
        className="c6 inline-editable"
      >
        <div
          className="inline-editable__edit-trigger"
          onClick={[Function]}
        >
          <span
            className="c4"
          >
            Especially this test!
          </span>
        </div>
      </div>
      <a
        className="c7"
      >
        February 12, 2020, 11:19 PM
      </a>
    </div>
  </div>
</div>
`;
